compiler/AST/astutil.cpp

```diff
@@ -169,26 +169,34 @@ void compute_call_sites() {
       fn->calledBy = new Vec<CallExpr*>();
   }
 
-  forv_Vec(CallExpr, call, gCallExprs) {
-    if (call->isEmpty() == true) {
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    for_SymbolSymExprs(se, fn) {
+      if (CallExpr* call = toCallExpr(se->parentExpr)) {
+        if (call->isEmpty() == true) {
+          assert(0); // not possible
 
-    } else if (isAlive(call) == false) {
+        } else if (isAlive(call) == false) {
+          assert(0); // not possible
 
-    } else if (FnSymbol* fn = call->resolvedFunction()) {
-      fn->calledBy->add(call);
+        } else if (fn == call->resolvedFunction()) {
+          fn->calledBy->add(call);
```

Comment:
(Michael and I discussed this off line and I am convinced, this post is for posterity)